vsintegration/src/FSharp.VS.FSI/sessions.fs

```diff
@@ -360,6 +361,49 @@ type FsiSession(sourceFile: string) =
     /// interrupt timeout in miliseconds
     let interruptTimeoutMS = 1000
 
+    let splitPidFile () =
+        let getTfmNumber (v: string) =
+            let arr = v.Split([| '.' |], 3)
+            match Single.TryParse(arr[0] + "." + arr[1]) with
+            | true, value -> Some value
+            | _ -> None
+
+        let frameworkVersion (frameworkDescription: string) =
+            let arr = frameworkDescription.Split([| ' ' |], 3)
+
+            match arr[0], arr[1] with
+            | ".NET", "Core" when arr.Length >= 3 -> getTfmNumber arr[2]
+            | ".NET", "Framework" when arr.Length >= 3 ->
+                if arr[ 2 ].StartsWith("4.8") then
+                    Some 4.8f
+                else
+                    Some 4.72f
+            | ".NET", "Native" -> None
+            | ".NET", _ when arr.Length >= 2 -> getTfmNumber arr[1]
+            | _ -> None
+
+        // When using .NET Core, allow up to 2 seconds to allow detection of process ID
+        // of inner process to complete on startup.  The only scenario where we ask for the process ID immediately after
+        // process startup is when the user clicks "Start Debugging" before the process has started.
+        for i in 0..10 do
+            if SessionsProperties.fsiUseNetCore && trueProcessPid.IsNone then
+                if File.Exists(trueProcessIdFile) then
+                    let lines = File.ReadAllLines trueProcessIdFile
+                    trueProcessPid <-
+                        if lines.Length <= 0 then
+                            None
+                        else
+                            Some (File.ReadAllText lines[0] |> int)
```

Comment:
How does this work? 🤔
`File.ReadAllText lines[0] |> int    ` = `trueProcessPid`
`File.ReadAllText lines[0] |> float32` = `trueFrameworkVersion`

We already read the file and the `lines` should contain the values, so I would expect it to be something like:
```fsharp
trueProcessPid       <- lines |> Seq.tryItem 0 |> Option.map int
trueFrameworkVersion <- lines |> Seq.tryItem 1 |> Option.map float32
```